internals/plan/plan_test.go

```diff
@@ -2045,3 +2071,87 @@ func (s *S) TestStartStopOrderMultipleLanes(c *C) {
 	c.Assert(lanes[1], DeepEquals, []string{"srv2"})
 	c.Assert(lanes[2], DeepEquals, []string{"srv3"})
 }
+
+// TestLayerBuiltinCompatible ensures layerBuiltins used in the plan package
+// reflects the same YAML fields as exposed in the Layer type.
+func (s *S) TestLayerBuiltinCompatible(c *C) {
+	fields := structYamlFields(plan.Layer{})
```

Comment:
It seems to me it'd make a simpler and better test if we just hard-coded this as `[]string{"summary", "description", "services", "checks", "log-targets"}`. If we get out of sync the test will tell us. Here we have a lot of machinery to do something we can easily do by hand (and who's testing the machinery :-).

Also, you might find `testutil.DeepUnsortedMatches` useful.

Reply:
I've changed the comment on the test function to indicate this is simply a change detector for ```plan.Layer``` and ```plan.Plan```. The ```builtinSections``` list of fields is just one source of hard-coded fields, so only testing that could be a too limiting check to really be of use. Overall, this test may not be that useful in practise, is test coverage is pretty good, and failing to update internal fields will almost certainly result in other tests failing.

I've updated the test to use ```testutil.DeepUnsortedMatches```. Happy to further simplify or remove this in a followup PR. Resolving for now.